src/lib/api/api-polyfill.ts

```diff
@@ -33,9 +33,26 @@ async function fetchHandler(
       // we get around that by renaming the file ext to .bin
       // see https://github.com/facebook/react-native/issues/27099
       // -prf
-      const newPath = reqBody.replace(/\.jpe?g$/, '.bin')
-      await RNFS.moveFile(reqBody, newPath)
-      reqBody = newPath
+
+      // On some platforms, moving this file is not possible. We will attempt to move it (this is optimal, since
+      // we don't create duplicates) and if there is an error, we will instead copy the file to the cache directory
+      const fileName = reqBody.split('/').pop() ?? ''
+      const newPath = `${cacheDirectory ?? ''}${fileName.replace(
+        /\.jpe?g$/,
+        '.bin',
+      )}`
+      try {
+        await moveAsync({
+          from: reqBody,
+          to: newPath,
+        })
+        reqBody = newPath
+      } catch (e) {
+        await copyAsync({
+          from: reqBody,
+          to: newPath,
+        })
+      }
```

Comment:
This is a fix for some devices preventing writing to the directory we are using. There are some users who report _never_ having been able to upload a photo. I've been able to finally reproduce this back to this function. The `moveAsync` or previously `RNFS.moveFile` would error out before the upload would succeed.

Here, we attempt to move first (that's the optimal way to do it, since it doesn't duplicate the file and fill up storage) but if we can't, we'll just copy it.